guides/hack/20-attributes/07-predefined-attributes.md

```diff
@@ -191,6 +191,26 @@ The interface type `IMemoizeParam` assists with memorizing objects passed to asy
 - If an exception is thrown, this is not memoized.
 - Functions with variadic parameters can not be memoized
 
+### Awaitables and Exceptions
+
+If the function returns an `Awaitable` (e.g. if it is an async function), the
+`Awaitable` is cached, instead of the final result. This usually is the expected
+behavior, however it can be surprising when exceptions are involved.
+
+Thrown exceptions are not memoized, showing by the increasing counter in this
+example:
+
+@@ predefined-attributes-examples/memoize-throw.hack @@
+
+However, if the the memoized function is `async`, the function does not
+technically throw - it returns an Awaitable that throws when resolved. This
+means that if the Awaitable is resolved multiple times, the same exception
+will be thrown every time; as the Awaitable is cached for `__Memoize`'d
+async functions, this results in the same exception being thrown every time,
+shown by the counter remaining 0 in this example:
```

Comment:
Could you break up this sentence?

> If the Awaitable is resolved multiple times, the same exception
will be thrown every time. For `__Memoize`'d async function, this results
> in the same exception being thrown each time.
>
> The counter remains 0 in this example: